### PR TITLE
[Disability Benefits Team 1] Use data we have instead to find icn

### PR DESCRIPTION
**Pull request by tommasina-va**
## Summary
- This work fixes a bug we found with 'undefined method 'icn' for nil' when Form526 submissions were being processed. The bug happened because the Backup Submission class could not find a user account. This fix uses the primary path's way of finding a submission's user account. 

## Related issue(s)

-https://app.zenhub.com/workspaces/disability-benefits-experience-team-1-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/94537
## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/sidekiq/form526_backup_submission_process/processor.rb [hunk 1]
```diff
@@ -442,7 +442,7 @@ def choose_provider(headers, provider, breakered: true)
           provider:,
           # this sends the auth headers and if we want the "breakered" or "non-breakered" version
           options: { auth_headers: headers, breakered: },
-          current_user: OpenStruct.new({ flipper_id: submission.user_uuid, icn: @user_account.icn }),
+          current_user: OpenStruct.new({ flipper_id: submission.user_uuid, icn: submission.user_account.icn }),
```
Comment (LindseySaari) [inline lib/sidekiq/form526_backup_submission_process/processor.rb:445]: Should we add a couple of safe navigation operators here to make the code more robust against nil values at any point in the chain? `submission&.user_account&.icn`

File: lib/sidekiq/form526_backup_submission_process/processor.rb [hunk 2]
```diff
@@ -84,6 +83,10 @@ def initialize(submission_id, docs = [], get_upload_location_on_instantiation: t
         determine_zip
       end
 
+      def submission_account(submission)
+        UserAccount.find_by(id: submission.user_account_id) || Account.lookup_by_user_uuid(submission.user_uuid)
```
Comment (aurora-a-k-a-lightning) [inline lib/sidekiq/form526_backup_submission_process/processor.rb:87]: ah, much more dependable! thanks for catching this

**Review by aurora-a-k-a-lightning — APPROVED** [review]
🚢

File: lib/sidekiq/form526_backup_submission_process/processor.rb (hunk 1 above)
Comment (tommasina-va) [reply lib/sidekiq/form526_backup_submission_process/processor.rb]: Good call. I changed the approach entirely to use the way the primary path sets the user account.

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**